qiskit/optimization/problems/quadratic_program.py

```diff
@@ -152,36 +153,139 @@ def _add_variable(self,
                       upperbound: Union[float, int] = INFINITY,
                       vartype: VarType = VarType.CONTINUOUS,
                       name: Optional[str] = None) -> Variable:
-        """Checks whether a variable name is already taken and adds the variable to list and index
-        if not.
+        return self._add_variables(lowerbound, upperbound, vartype, name)[1][0]
+
+    def _add_variables(self,
+                       lowerbound: Union[float, int] = 0,
+                       upperbound: Union[float, int] = INFINITY,
+                       vartype: VarType = VarType.CONTINUOUS,
+                       name: Optional[str] = None,
+                       key_format: Optional[str] = None,
+                       keys: Union[int, Sequence] = 1) -> Tuple[List[str], List[Variable]]:
+        if isinstance(keys, int) and keys < 1:
+            raise QiskitOptimizationError(
+                "Cannot create non-positive number of variables: {}".format(keys))
+        if not name and key_format:
+            name = 'x'
+        if name and not key_format:
+            if isinstance(keys, int) and keys > 1 or isinstance(keys, Sequence) and len(keys) > 0:
+                key_format = '{}'
+            else:
+                key_format = ''
+                if name in self._variables_index:
+                    raise QiskitOptimizationError("Variable name already exists: {}".format(name))
+        if not name and not key_format:
+            name = 'x'
+            key_format = '{}'
+        if '{{}}' in key_format:
+            raise QiskitOptimizationError(
+                "Formatter cannot contain nested substitutions: {}".format(key_format))
+        substitution_count = key_format.count('{}')
+        if substitution_count > 1:
+            raise QiskitOptimizationError(
+                "Formatter cannot contain more than one substitution: {}".format(key_format))
+
+        names = []
+        variables = []
+        if isinstance(keys, Sequence):
+            for key in keys:
+                indexed_name = name + key_format.format(str(key))
+                if indexed_name in self._variables_index:
+                    raise QiskitOptimizationError(
+                        "Variable name already exists: {}".format(indexed_name))
+                names.append(indexed_name)
+                variables.append(
+                    self._create_var_update_index(lowerbound, upperbound, vartype, indexed_name))
+        else:
+            k = self.get_num_vars()
+            for _ in range(keys):
+                while name + key_format.format(k) in self._variables_index:
+                    k += 1
+                indexed_name = name + key_format.format(k)
+                names.append(indexed_name)
+                variables.append(
+                    self._create_var_update_index(lowerbound, upperbound, vartype, indexed_name))
```

Comment:
Just curious, if this part replace with a `for key in keys` loop and then inside the loop add `if` statement, would be better and easier to read? I just don't like the fact the logic in the loops is similar.